azure_jumpstart_ag/artifacts/PowerShell/AgLogonScript.ps1

```diff
@@ -260,18 +260,31 @@ if ($httpResponse.StatusCode -ne 200){
 
 ### BELOW IS AN ALTERNATIVE APPROACH TO IMPORT DASHBOARD USING README INSTRUCTIONS
 
-$agDir = $AgConfig.AgDirectories["AgDir"]
-$adxEndPoint = (az kusto cluster show --name $adxClusterName --resource-group $resourceGroup --query "uri" -o tsv --only-show-errors)
-if ($null -ne $adxEndPoint -and $adxEndPoint -ne "") {
-    $ordersDashboardBody = (Invoke-WebRequest -Method Get -Uri "$templateBaseUrl/artifacts/adx-dashboard-orders-payload.json").Content -replace '{{ADX_CLUSTER_URI}}', $adxEndPoint -replace '{{ADX_CLUSTER_NAME}}', $adxClusterName
-    Set-Content -Path "$agDir\adx-dashboard-orders-payload.json" -Value $ordersDashboardBody -Force -ErrorAction Ignore
-    $iotSensorsDashboardBody = (Invoke-WebRequest -Method Get -Uri "$templateBaseUrl/artifacts/adx-dashboard-iotsensor-payload.json") -replace '{{ADX_CLUSTER_URI}}', $adxEndPoint -replace '{{ADX_CLUSTER_NAME}}', $adxClusterName
-    Set-Content -Path "$agDir\adx-dashboard-iotsensor-payload.json" -Value $iotSensorsDashboardBody -Force -ErrorAction Ignore
-}
-else {
-    Write-Host "[$(Get-Date -Format t)] ERROR: Unable to find Azure Data Explorer endpoint from the cluster resource in the resource group."
+$adxDashBoardsDir = $AgConfig.AgDirectories["AgAdxDashboards"]
+$dataEmulatorDir = $AgConfig.AgDirectories["AgDataEmulator"]
+$kustoCluster = Get-AzKustoCluster -ResourceGroupName $resourceGroup -Name $adxClusterName
+if ($null -ne $kustoCluster) {
+    $adxEndPoint = $kustoCluster.Uri
+    if ($null -ne $adxEndPoint -and $adxEndPoint -ne "") {
+        $ordersDashboardBody = (Invoke-WebRequest -Method Get -Uri "$templateBaseUrl/artifacts/adx_dashboards/adx-dashboard-orders-payload.json").Content -replace '{{ADX_CLUSTER_URI}}', $adxEndPoint -replace '{{ADX_CLUSTER_NAME}}', $adxClusterName
+        Set-Content -Path "$adxDashBoardsDir\adx-dashboard-orders-payload.json" -Value $ordersDashboardBody -Force -ErrorAction Ignore
+        $iotSensorsDashboardBody = (Invoke-WebRequest -Method Get -Uri "$templateBaseUrl/artifacts/adx_dashboards/adx-dashboard-iotsensor-payload.json") -replace '{{ADX_CLUSTER_URI}}', $adxEndPoint -replace '{{ADX_CLUSTER_NAME}}', $adxClusterName
+        Set-Content -Path "$adxDashBoardsDir\adx-dashboard-iotsensor-payload.json" -Value $iotSensorsDashboardBody -Force -ErrorAction Ignore
+    }
+    else {
+        Write-Host "[$(Get-Date -Format t)] ERROR: Unable to find Azure Data Explorer endpoint from the cluster resource in the resource group."
+    }
 }
 
+# Download DataEmulator.zip into Agora folder and unzip
+$emulatorPath = "$dataEmulatorDir\DataEmulator.zip"
+Invoke-WebRequest -Method Get -Uri "$templateBaseUrl/artifacts/data_emulator/DataEmulator.zip" -OutFile $emulatorPath
+
+# Unzip DataEmulator.zip to copy DataEmulator exe and config file to generate sample data for dashboards
+if ([System.IO.File]::Exists($emulatorPath)) {
```

Comment:
consider Test-Path instead of the .NET call here for future use

Reply:
Good advise, will change in this release. :)

Reply:
Updated